internal/controller/controllers/infraenv_controller.go

```diff
@@ -244,6 +244,7 @@ func (r *InfraEnvReconciler) ensureISO(ctx context.Context, log logrus.FieldLogg
 	if err != nil {
 		if errors.Is(err, gorm.ErrRecordNotFound) {
 			Requeue = true
+			err = errors.Errorf("cluster does not exist: %s", clusterDeployment.Name)
```

Comment:
I think we should also add a mention of AgentClusterInstall.
Maybe if clusterDeployment.Spec.ClusterInstallRef is not nil, add it also to the message ("Verify AgentClusterInstall %s exists", clusterDeployment.Spec.ClusterInstallRef.Name)